starknet_py/tests/e2e/account/account_test.py

```diff
@@ -98,6 +99,48 @@ async def test_estimate_fee_for_declare_transaction(account, map_compiled_contra
     )
 
 
+@pytest.mark.asyncio
+async def test_account_estimate_fee_for_declare_transaction(
+    account, map_compiled_contract
+):
+    declare_tx = await account.sign_declare_v1(
+        compiled_contract=map_compiled_contract, max_fee=MAX_FEE
+    )
+
+    estimated_fee = await account.estimate_fee(tx=declare_tx)
+
+    assert estimated_fee.unit == PriceUnit.WEI
+    assert isinstance(estimated_fee.overall_fee, int)
+    assert estimated_fee.overall_fee > 0
+    assert (
+        estimated_fee.gas_consumed * estimated_fee.gas_price
+        == estimated_fee.overall_fee
+    )
+
+
+@pytest.mark.asyncio
+async def test_account_estimate_fee_for_transactions(account, map_compiled_contract):
+    nonce = await account.get_nonce(block_hash="pending")
+    print(nonce)
+    declare_tx = await account.sign_declare_v1(
+        compiled_contract=map_compiled_contract, max_fee=MAX_FEE, nonce=nonce
+    )
+    declare_tx2 = await account.sign_declare_v1(
+        compiled_contract=map_compiled_contract, max_fee=MAX_FEE, nonce=(nonce + 1)
+    )
+
+    estimated_fee = await account.estimate_fee(tx=[declare_tx, declare_tx2])
```

Comment:
Can we have two distinct transaction types? I believe it would be even better if the second transaction were in version 3.